src/libmongoc/src/mongoc/mongoc-topology-private.h

```diff
@@ -46,6 +46,28 @@ typedef enum {
 struct _mongoc_background_monitor_t;
 struct _mongoc_client_pool_t;
 
+typedef enum { MONGOC_RR_SRV, MONGOC_RR_TXT } mongoc_rr_type_t;
```

Comment:
`mongoc_rr_type_t` and the `_mongoc_rr_data_t` struct were moved from mongoc-client-private.h to add the mock function pointer to `mongoc_topology_t`. I think all of the SRV / TXT lookup code should be moved out of `mongoc-client.c` since it is not used by any of the `mongoc_client_t` functions. I left a TODO comment in `mongoc-client.c` to avoid unnecessary code movement.